service/src/java/org/apache/hive/service/server/KillQueryImpl.java

```diff
@@ -116,9 +117,21 @@ public static void killChildYarnJobs(Configuration conf, String tag, String doAs
 
   private static boolean isAdmin() {
     boolean isAdmin = false;
-    if (SessionState.get().getAuthorizerV2() != null) {
+    SessionState ss = SessionState.get();
+    if(!HiveConf.getBoolVar(ss.getConf(), HiveConf.ConfVars.HIVE_AUTHORIZATION_ENABLED)) {
+      // If authorization is disabled, hs2 process owner should have kill privileges
       try {
-        SessionState.get().getAuthorizerV2()
+        String currentUser = ss.getUserName();
```

Comment:
Can we try and add an unit test that hits this flow (if not already exist)?

Reply:
org.apache.hive.jdbc.TestJdbcWithMiniLlapArrow#testKillQueryByTagAdmin

Reply:
Added Tests